tests/v2/validation/nodescaling/replace.go

```diff
@@ -24,46 +24,50 @@ const (
 	clusterLabel                 = "cluster.x-k8s.io/cluster-name"
 )
 
-func MatchNodeToRole(t *testing.T, client *rancher.Client, clusterID string, isEtcd bool, isControlPlane bool, isWorker bool) (int, *management.Node) {
+func MatchNodeToRole(t *testing.T, client *rancher.Client, clusterID string, isEtcd bool, isControlPlane bool, isWorker bool) (int, []management.Node) {
 	machines, err := client.Management.Node.List(&types.ListOpts{Filters: map[string]interface{}{
 		"clusterId": clusterID,
 	}})
 	require.NoError(t, err)
+
 	numOfNodes := 0
-	lastMatchingNode := &management.Node{}
+	matchingNodes := []management.Node{}
 
 	for _, machine := range machines.Data {
 		if machine.Etcd == isEtcd && machine.ControlPlane == isControlPlane && machine.Worker == isWorker {
-			lastMatchingNode = &machine
+			matchingNodes = append(matchingNodes, machine)
 			numOfNodes++
 		}
 	}
-	require.NotEmpty(t, lastMatchingNode.NodeName, "matching node name is empty")
-	return numOfNodes, lastMatchingNode
+	require.NotEmpty(t, matchingNodes, "matching node name is empty")
+
+	return numOfNodes, matchingNodes
 }
 
 // ReplaceNodes replaces the last node with the specified role(s) in a k3s/rke2 cluster
 func ReplaceNodes(t *testing.T, client *rancher.Client, clusterName string, isEtcd bool, isControlPlane bool, isWorker bool) {
 	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
 	require.NoError(t, err)
-	numOfNodesBeforeDeletion, nodeToDelete := MatchNodeToRole(t, client, clusterID, isEtcd, isControlPlane, isWorker)
+	numOfNodesBeforeDeletion, nodesToDelete := MatchNodeToRole(t, client, clusterID, isEtcd, isControlPlane, isWorker)
 
-	machineToDelete, err := client.Steve.SteveType(machineSteveResourceType).ByID("fleet-default/" + nodeToDelete.Annotations[machineSteveAnnotation])
-	require.NoError(t, err)
+	for i := range nodesToDelete {
+		machineToDelete, err := client.Steve.SteveType(machineSteveResourceType).ByID("fleet-default/" + nodesToDelete[i].Annotations[machineSteveAnnotation])
+		require.NoError(t, err)
 
-	logrus.Info("Deleting, " + nodeToDelete.Name + " node..")
-	err = client.Steve.SteveType(machineSteveResourceType).Delete(machineToDelete)
-	require.NoError(t, err)
+		logrus.Infof("Replacing node: " + nodesToDelete[i].NodeName)
```

Comment:
Super nit, I think logrus.Info is more appropriate here.. since you aren't inserting variables into the string